Experiences/Experiences/View Controllers/CameraViewController.swift

```diff
@@ -0,0 +1,265 @@
+            self.present(noMicAlert, animated: true)
+        }
+        cameraView.videoPlayerLayer.videoGravity = .resizeAspectFill
+        
+        
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTapGesture(_:)))
+        view.addGestureRecognizer(tapGesture)
+        
+        
+    }
+    
+    @objc func handleTapGesture(_ tapGesture: UITapGestureRecognizer) {
+        if tapGesture.state == .ended {
+            replayRecording()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        captureSession.startRunning()
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        captureSession.stopRunning()
+    }
+    
+    private func updateViews() {
+        recordButton.isSelected = fileOutput.isRecording
+    }
+    
+    
+    // MARK: - Camera & Microphone Set-Up
+    
+    private func setUpCamera() {
+        guard let camera = camera, let microphone = microphone else { return }
+        
+        captureSession.beginConfiguration()
+        
+        guard let cameraInput = try? AVCaptureDeviceInput(device: camera) else {
+            preconditionFailure("Cannot get camera input")
+        }
+        
+        guard let microphoneInput = try? AVCaptureDeviceInput(device: microphone) else {
+            preconditionFailure("Cannot get microphone input")
+        }
+        
+        // Add Video input
+        guard captureSession.canAddInput(cameraInput) else {
+            preconditionFailure("Unable to add camera input")
+        }
+        captureSession.addInput(cameraInput)
+        
+        // Add audio input
+        guard captureSession.canAddInput(microphoneInput) else {
+            preconditionFailure("Unable to add microphone input")
+        }
+        captureSession.addInput(microphoneInput)
+        
+        // Set video present
+        if captureSession.canSetSessionPreset(.hd1920x1080) {
+            captureSession.sessionPreset = .hd1920x1080
+        }
+        
+        // Check if can output & add output
+        guard captureSession.canAddOutput(fileOutput) else {
+            preconditionFailure("Cannot write to disk")
+        }
+        captureSession.addOutput(fileOutput)
+        
+        captureSession.commitConfiguration()
+        cameraView.session = captureSession
+    }
+    
+    private func bestCamera() {
+        if let device = AVCaptureDevice.default(.builtInUltraWideCamera, for: .video, position: .back) {
+            camera = device
+        }
+        if let device = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back) {
+            camera = device
+        }
+    }
+    
+    private func bestMicrophone() {
+        if let device = AVCaptureDevice.default(for: .audio) {
+            microphone = device
+        }
+    }
+    
+    
+    //MARK: - Video Recording Functions
+       
```

Comment:
btw: another way to write this if else statement:
 fileOutput.isRecording == true ? fileOutput.stopRecording() : fileOutput.startRecording(to: newRecordingURL(), recordingDelegate: self)